vm/src/obj/objtuple.rs

```diff
@@ -234,6 +235,10 @@ impl PyTuple {
         vm: &VirtualMachine,
     ) -> PyResult<PyTupleRef> {
         let elements = if let OptionalArg::Present(iterable) = iterable {
+            if cls.is(&vm.ctx.tuple_type()) && iterable.lease_class().is(&vm.ctx.tuple_type()) {
+                return PyTupleRef::try_from_object(vm, iterable.clone());
```

Comment:
`try_from_object` checks type matches, but we already know iterable is compatible with return value.
Using `PyObject::into_pyref()` will remove the typecheck cost.
```suggestion
                return iterable.into_pyref();
```

Reply:
That wouldn't work, that only works for `PyRc<PyObject<PyTuple>>`, not for any PyObjectRef. I think this is fine for now, we could maybe add a method downcast_exact to PyObjectRef if we want this.

Reply:
Ah, ok. In that case, I think changing return type to `PyObjectRef` would be the best. If it is too much, to track this issue in future, adding a comment including link to this issue will be great.